apps/auto-email-sender-new/src/index.ts

```diff
@@ -1,15 +1,107 @@
-import dotenv from 'dotenv'
 import { rolesRenderer } from './roles-renderer'
 import { subsToQueue } from './subs-to-queue'
-import { assignRoles } from './roles-assigner'
+import { assignRoles } from './roles-assigner-new'
 import { emailPreRender } from './email-pre-render'
 import { composeEmail } from './email-composer'
 import { emailSender } from './email-sender'
+import { getSupabaseClient } from 'db'
+import { Parser } from '@json2csv/plainjs'
+import fs from 'fs'
+import { exec } from 'child_process'
+import { promisify } from 'util'
+import path from 'path'
+import axios from 'axios'
 
-dotenv.config()
+const generateCsv = async (data: any, filePath: string) => {
+  const parser = new Parser()
+  const csv = parser.parse(data.data)
+  const absolutePath = path.resolve(__dirname, filePath)
+  fs.writeFileSync(absolutePath, csv)
+}
+
+const runCommand = promisify(exec)
+
+const checkMatchRolesUp = async (
+  url: string,
+  maxAttempts = 10,
+  interval = 5000
+) => {
+  const delay = (ms: number) =>
+    new Promise((resolve) => setTimeout(resolve, ms))
+  let attempt = 0
+  while (attempt < maxAttempts) {
+    try {
+      const response = await axios.get(url)
+      if (response.status === 200) {
+        console.log('match_roles is up.')
+        return
+      }
+    } catch (error) {
+      console.error(`Error checking match_roles: ${error}`)
+    }
+    attempt++
+    console.log(
+      `Attempt ${attempt} failed. Retrying in ${interval / 1000} seconds...`
+    )
+    await delay(interval)
+  }
+  throw new Error(`match_roles did not start after ${maxAttempts} attempts.`)
+}
 
 export const init = async () => {
+  const supabaseClient = getSupabaseClient()
+
   try {
+    const roles = await supabaseClient
+      .from('Roles')
+      .select('*')
+      .eq('ready', true)
+
+    await generateCsv(roles, './match_roles/data/roles.csv')
+
+    const skills = await supabaseClient.from('Skills').select('*')
+
+    await generateCsv(skills, './match_roles/data/skills.csv')
+
+    const matchRolesPath = path.join(__dirname, 'match_roles')
+
+    console.log('Starting match_roles')
+
+    console.log('Generating matchenv...')
+    await runCommand('bash -c "python3 -m venv matchenv"', {
+      cwd: matchRolesPath,
+    })
+
+    console.log('Install dependencies...')
+    await runCommand(
+      'bash -c "source matchenv/bin/activate && pip install -r requirements.txt"',
+      {
+        cwd: matchRolesPath,
+      }
+    )
+
+    console.log('Generating onehot from skills...')
+    await runCommand(
+      'bash -c "source matchenv/bin/activate && cd src/train && python3 onehot.py --entity skills"',
+      {
+        cwd: matchRolesPath,
+      }
+    )
+
+    console.log('Starting match_roles')
+    runCommand(
+      'bash -c "source matchenv/bin/activate && cd src && uvicorn --host 0.0.0.0 main:app"',
+      {
+        cwd: matchRolesPath,
+      }
+    )
+
```

Comment:
Please extract this to a single function called `setupMatchRoles()` to cleanup this file.